logging/telemetryhook.go

```diff
@@ -73,22 +80,31 @@ func createAsyncHook(wrappedHook logrus.Hook, channelDepth uint, maxQueueDepth i
 
 func (hook *asyncTelemetryHook) appendEntry(entry *logrus.Entry) {
 	hook.Lock()
+	defer hook.Unlock()
+	// TODO: If there are errors at startup, before the telemetry URI is set, this can fill up. Should we prioritize
+	//       startup / heartbeat events?
 	if len(hook.pending) >= hook.maxQueueDepth {
 		hook.pending = hook.pending[1:]
 		hook.wg.Done()
 	}
 	hook.pending = append(hook.pending, entry)
-	hook.Unlock()
 }
 
-func (hook *asyncTelemetryHook) waitForEvent() bool {
-	select {
-	case <-hook.quit:
-		return false
-	case entry := <-hook.entries:
-		hook.appendEntry(entry)
-		return true
+func (hook *asyncTelemetryHook) waitForEventAndReady() bool {
+	ticker := time.NewTicker(time.Second)
+
+	for !hook.ready || len(hook.pending) == 0 {
```

Comment:
this would cause a data race.
The ready and pending should be gated behind the hook lock.